src/chunk_list.cpp

```diff
@@ -721,32 +736,26 @@ void chunk_swap_lines(chunk_t *pc1, chunk_t *pc2)
 } // chunk_swap_lines
 
 
-/**
- * Gets the next non-vbrace chunk
- */
-chunk_t *chunk_get_next_nvb(chunk_t *cur, chunk_nav_t nav)
+chunk_t *chunk_get_next_nvb(chunk_t *cur, const chunk_nav_t nav)
 {
-   chunk_t *pc = cur;
+   return(chunk_get_nvb(cur, nav, true));       // \todo check this
+}
 
-   do
-   {
-      pc = chunk_get_next(pc, nav);
-   } while (chunk_is_vbrace(pc));
-   return(pc);
+
+chunk_t *chunk_get_prev_nvb(chunk_t *cur, const chunk_nav_t nav)
+{
+   return(chunk_get_nvb(cur, nav, false));
 }
 
 
-/**
- * Gets the prev non-vbrace chunk
- */
-chunk_t *chunk_get_prev_nvb(chunk_t *cur, chunk_nav_t nav)
+static chunk_t *chunk_get_nvb(chunk_t *cur, const chunk_nav_t nav, const bool dir)
 {
    chunk_t *pc = cur;
 
    do
    {
-      pc = chunk_get_prev(pc, nav);
-   } while (chunk_is_vbrace(pc));
+      pc = (dir == true) ? chunk_get_next(pc, nav) : chunk_get_prev(pc, nav);
```

Comment:
So for each token, instead of calling a function directly we're also going to check a condition ? Do you think the code is still going to remain performant ?